examples/cloud/gettingStartedConsole.php

```diff
@@ -47,44 +47,45 @@
     {
-        // Use the command line args to get the resource key if present.
-        // Otherwise, get it from the environment variable.
-        $resourceKey = isset($argv) && count($argv) > 0 ? $argv[0] : ExampleUtils::getResourceKey();
-        
-        // Configure a logger to output to the console.
+        // Configure a logger to output to the console
         $logger = new Logger("info");
 
         // Load the configuration file
         $config = json_decode(file_get_contents(__DIR__."/gettingStartedConsole.json"), true);
-        // Get the resource key setting from the config file. 
-        $resourceKeyFromConfig = ExampleUtils::getResourceKeyFromConfig($config);
-        $configHasKey = empty($resourceKeyFromConfig) == false && strpos($resourceKeyFromConfig, "!!") !== 0;
 
-        // If no resource key is specified in the config file then override it with the key
-        // from the environment variable / command line. 
-        if ($configHasKey === false)
-        {
-            ExampleUtils::setResourceKeyInConfig($config, $resourceKey);
+        // Get the resource key from command line args
+        $resourceKey = ExampleUtils::getResourceKeyFromCliArgs($argv);
+
+        // Otherwise, get the resource key from the environment variable
+        if (empty($resourceKey)) {
+            $resourceKey = ExampleUtils::getResourceKeyFromEnv();
         }
 
-        if (empty($resourceKey) == false)
-        {
-            (new GettingStartedConsole())->run($config, $logger, ['fiftyone\\pipeline\\devicedetection\\examples\\cloud\\classes\\ExampleUtils', 'output']);
+        // Otherwise, get the resource key from the config file
+        if (empty($resourceKey)) {
+            $resourceKey = ExampleUtils::getResourceKeyFromConfig($config);
         }
-        else
-        {
-            $logger->log("error",
-                "No resource key specified in environment variable " .
-                "'".ExampleUtils::RESOURCE_KEY_ENV_VAR."'. The 51Degrees " .
-                "cloud service is accessed using a 'ResourceKey'. " .
+        
+        if (empty($resourceKey)) {
+            $message = "No resource key specified in CLI args, environment variable '" .
+                ExampleUtils::RESOURCE_KEY_ENV_VAR . "', or configuration file." . PHP_EOL .
+                "The 51Degrees cloud service is accessed using a 'ResourceKey'. " .
                 "For more detail see " .
                 "http://51degrees.com/documentation/4.3/_info__resource_keys.html. " .
                 "A resource key with the properties required by this " .
                 "example can be created for free at " .
                 "https://configure.51degrees.com/g3gMZdPY. " .
                 "Once complete, populated the environment variable " .
-                "mentioned at the start of this message with the key.");
+                "mentioned at the start of this message with the key.";
+            
+            $logger->log("error", $message);
```

Comment:
this is nice, but I see a duplication of these 3 lines.  could we introduce a funciton in the ExampleUtils like: 
```php
ExampleUtils::logError($logger, $message) { 
  $logger->log..
  echo..
  exit..
}
```

Reply:
Lines replaced with `logErrorAndExit()`.